apps/explorer/src/components/ownedobjects/views/FieldItem.tsx

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { type SuiMoveNormalizedType } from '@mysten/sui.js';
+
+import { SyntaxHighlighter } from '~/components/SyntaxHighlighter';
+import {
+    extractSerializationType,
+    getFieldTypeValue,
+    FieldValueType,
+} from '~/components/ownedobjects/utils';
+import { AddressLink, ObjectLink } from '~/ui/InternalLink';
+import { Link } from '~/ui/Link';
+import { Text } from '~/ui/Text';
+
+interface FieldItemProps<T> {
+    value: T;
+    type: SuiMoveNormalizedType;
+    truncate?: boolean;
+}
+
+export function FieldItem<T>({
+    value,
+    type,
+    truncate = false,
+}: FieldItemProps<T>) {
+    // for object types, use SyntaxHighlighter
+    if (typeof value === 'object') {
+        return (
+            <SyntaxHighlighter
+                code={JSON.stringify(value, null, 2)}
+                language="json"
+            />
+        );
+    }
+
+    const normalizedType = extractSerializationType(type);
+    const moduleName = getFieldTypeValue(normalizedType, FieldValueType.MODULE);
+    const address = getFieldTypeValue(normalizedType, FieldValueType.ADDRESS);
+    const name = getFieldTypeValue(normalizedType, FieldValueType.NAME);
+
+    if (typeof value === 'string' && normalizedType === 'Address') {
+        return (
+            <div className="break-all">
+                <AddressLink address={value} noTruncate={!truncate} />
+            </div>
+        );
+    }
+
+    const isObjectId = ['0x2::object::UID', '0x2::object::ID'].includes(
+        `${address}::${moduleName}::${name}`
+    );
+    if (typeof value === 'string' && isObjectId) {
```

Comment:
Can the value type actually not be a string?

Reply:
Yes, it could be a Boolean or Number

Reply:
But I imagine if the type is an object id, then it could be right? Like the typescript type might not refine correctly but on-chain I imagine you can't store a number in a UID

Reply:
Made change to only check against the normalizedType